### PR TITLE
Added release notes for 1.4.0

### LINKED ISSUE #1394: Create Data Prepper 1.4.0 Release Notes
Create Data Prepper 1.4.0 Release Notes

All changes should be available at:

https://github.com/opensearch-project/data-prepper/milestone/5?closed=1

Please note that we need to clarify in the Release Notes which OTel Metric types are currently supported and which are not.

### PR DESCRIPTION
**Pull request by asifsmohammed**
Signed-off-by: Asif Sohail Mohammed <nsifmoh@amazon.com>

### Issues Resolved
Resolves #1394 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thanks! I have a few recommendations to push down some of the development-related items.

File: release/release-notes/data-prepper.release-notes-1.4.0.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+### Features
+* Support metrics ingestion ([#1154](https://github.com/opensearch-project/data-prepper/pull/1154), [#1271](https://github.com/opensearch-project/data-prepper/pull/1271))
+* Support OpenSearch 2.0
+  * Update to use opensearch-java client instead of Rest High Level Client for bulk requests. ([#1381](https://github.com/opensearch-project/data-prepper/pull/1381))
+  * OpenSearch build files clean-up ([#1315](https://github.com/opensearch-project/data-prepper/pull/1315))
+
+### Enhancements
+* Migrate Trace Analytics plugins to Event Model ([#1216](https://github.com/opensearch-project/data-prepper/pull/1216), [#1223](https://github.com/opensearch-project/data-prepper/pull/1223), [#1224](https://github.com/opensearch-project/data-prepper/pull/1224), [#1220](https://github.com/opensearch-project/data-prepper/pull/1220), [#1237](https://github.com/opensearch-project/data-prepper/pull/1237), [#1241](https://github.com/opensearch-project/data-prepper/pull/1241), [#1239](https://github.com/opensearch-project/data-prepper/pull/1241))
+* Set 30 minute timeout to release process GitHub actions ([#1392](https://github.com/opensearch-project/data-prepper/pull/1392))
```
[the remark this answers is not in the document]
Comment (asifsmohammed) [reply release/release-notes/data-prepper.release-notes-1.4.0.md:13]: I'm going to move them to maintenance and stick with our existing categories. Maybe we can have some guidelines on this.

File: release/release-notes/data-prepper.release-notes-1.4.0.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+### Infrastructure
+* Added needs-documentation label ([#1373](https://github.com/opensearch-project/data-prepper/pull/1373))
+* Upload and publish JUnit test reports for some tests ([#1336](https://github.com/opensearch-project/data-prepper/pull/1336))
+
+### Documentation
+* Performance tests documentation ([#1309](https://github.com/opensearch-project/data-prepper/pull/1309))
+* Added custom metrics in otel-trace-source ([#1246](https://github.com/opensearch-project/data-prepper/pull/1246))
+
+### Maintenance
+* Support for OpenSearch 2.0 (previously supported versions of OpenSearch and OpenDistro are still supported)
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-1.4.0.md:20]: Sorry for the confusion on this. I do believe that this support should be an Enhancement. It has a direct impact on end-users. But, the two sub-bullets are more low-level details.

Comment (dlvenable) [reply release/release-notes/data-prepper.release-notes-1.4.0.md:20]: This should be in the Enhancements:

>  Support for OpenSearch 2.0 (previously supported versions of OpenSearch and OpenDistro are still supported)

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: release/release-notes/data-prepper.release-notes-1.4.0.md (hunk 2 above)
Comment (asifsmohammed) [reply release/release-notes/data-prepper.release-notes-1.4.0.md:20]: No problem, Thanks for the feedback David.

**Review by dlvenable — APPROVED** [review]

**Pull request merged**